microsoft/electionguard-ui
# 🔑 Fix Join Key Setup

### Issue

Fixes #128 

### Description
This fixes the joint key setup wizard so you can now complete it from beginning to end and it will create a key.  It also:

- Reworks the UI to make it feel clean and simple
- Removes one of the two confirmation pages
- Populates the assign guardians datagrid with a list of users from the back-end API
- Solves bearer auth with the back-end endpoints that require it (get users)
- Detects 401 token expiration requests and redirects the user to the login screen when they occur
- Fixes the back buttons so that data is preserved

### Testing

1. Log in
2. Click Setup Joint Keys

Expected: a page like this:

![image](https://user-images.githubusercontent.com/1769905/153657515-5edb4161-8317-4b31-8bbd-fdf0196f6ba6.png)

3. Fill in some data and click Next

Expected: Users in the system should be listed

4. Select some

Expected: they should select like this;

![image](https://user-images.githubusercontent.com/1769905/153658757-7bae3cb5-5d3e-4df6-8465-637a7e461d67.png)

5. Click Next

Expected: you should see a confirmation page with your choices like this:

![image](https://user-images.githubusercontent.com/1769905/153658960-46af37d5-7111-4bbd-a724-98da054c7fcd.png)

6. Click Submit
7. Setup Election and go to step 2

Your new key should be on the dropdown list:

![image](https://user-images.githubusercontent.com/1769905/153659284-78af1e76-b5eb-44ee-9a91-44d81f18df0a.png)
